library/core/src/main/java/com/google/android/exoplayer2/video/VideoFrameReleaseTimeHelper.java

```diff
@@ -99,6 +102,37 @@ public void enable() {
   public void disable() {
     if (useDefaultDisplayVsync) {
       vsyncSampler.removeObserver();
+      unregisterDisplayListener();
+    }
+  }
+
+  private void registerDisplayListener() {
+    if (displayListener == null && context != null &&
+     Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+      DisplayManager displayManager = context.getSystemService(DisplayManager.class);
+      if (displayManager != null) {
+        displayListener = new DefaultDisplayListener(context);
+        displayManager.registerDisplayListener(displayListener, null);
+      }
+    }
+  }
+
+  private void unregisterDisplayListener() {
+    if (context != null && displayListener != null &&
+     Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+      DisplayManager displayManager = context.getSystemService(DisplayManager.class);
+      if (displayManager != null) {
```

Comment:
Hm - You really don't want to get here and find displayManager is null. Maybe you should defensively stash it in a variable in registerDisplayListener, so you have guaranteed access to it.

Reply:
Or just get the displayManager once in the constructor, and have it be final.